src/directives/unsafe-svg.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {isPrimitive} from '../lib/parts.js';
+import {directive, NodePart, Part, reparentNodes} from '../lit-html.js';
+
+interface PreviousValue {
+  readonly value: unknown;
+  readonly fragment: DocumentFragment;
+}
+
+// For each part, remember the value that was last rendered to the part by the
+// unsafeSVG directive, and the DocumentFragment that was last set as a value.
+// The DocumentFragment is used as a unique key to check if the last value
```

Comment:
It's not clear why the document fragment is used. In what case does is it useful?

Reply:
See the 'renders after other values' test. If you render an `unsafeSVG()`, then any other value, then `unsafeSVG()` again with the same parameter, we need to re-render to the part, even though the parameter is the same. Checking that the current value is the right DocumentFragment as part of the dirty-check allows this.